lib/src/components/NxList/NxList.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef, MutableRefObject, useCallback, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import NxListText from './NxListText';
+import NxListSubtext from './NxListSubtext';
+import NxListActions from './NxListActions';
+import NxListButtonItem from './NxListButtonItem';
+import NxListLinkItem from './NxListLinkItem';
+import NxListItem from './NxListItem';
+import NxListDescriptionTerm from './NxListDescriptionTerm';
+import NxListDescription from './NxListDescription';
+import { NxH3 } from '../SimpleComponents';
+import { splitOutFirst } from '../../util/childUtil';
+import { NxListProps, NxLoadError, NxLoadingSpinner } from '../..';
+import useMutationObserver from '@rooks/use-mutation-observer';
+import { NxListComponent, nxListPropTypes } from './types';
```

Comment:
We keep external lib imports at the top, separated by a blank line.  Additionally it can make sense to group imports from within the same directory separately from those outside.  Also don't import from `index.ts` within a component file, that creates a circular dependency.  It might work, but it's best avoided.

```suggestion
import useMutationObserver from '@rooks/use-mutation-observer';

import { NxH3 } from '../SimpleComponents';
import NxLoadingSpinner from '../NxLoadingSpinner/NxLoadingSpinner';
import NxLoadError from '../NxLoadError/NxLoadError';
import { splitOutFirst } from '../../util/childUtil';

import NxListText from './NxListText';
import NxListSubtext from './NxListSubtext';
import NxListActions from './NxListActions';
import NxListButtonItem from './NxListButtonItem';
import NxListLinkItem from './NxListLinkItem';
import NxListItem from './NxListItem';
import NxListDescriptionTerm from './NxListDescriptionTerm';
import NxListDescription from './NxListDescription';
import { NxListProps, NxListComponent, nxListPropTypes } from './types';
```